src/main/java/views/add_edit_category_views/CategoryP.java

```diff
@@ -0,0 +1,42 @@
+package views.add_edit_category_views;
+
+import use_cases.add_edit_category_use_case.CategoryOB;
+import use_cases.add_edit_category_use_case.CategoryOD;
+
+/**
+ * CategoryP notifies success/fail to add or edit category, implements CategoryOB.
+ * Returns CategoryOD when execution is successful.
+ * Returns String error message when failed.
+ */
+public class CategoryP implements CategoryOB {
+    /**
+     * Returns a CategoryOD notifying success add.
+     * @return String success add message.
+     */
+    @Override
+    public CategoryOD success_add() {
+        String successAdd = "You have added a new category!";
+        return new CategoryOD(successAdd);
+    }
```

Comment:
The CategoryOD objects should be created in the use case interactor and passed in to CategoryOB as a parameter. These shouldn't create new CategoryOD objects.